kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -351,21 +351,38 @@ def retrieve(
                 url, params=query_params, headers=self.service.headers
             )
             response.raise_for_status()
+        except HTTPError as er:
+            nexus_path = f"{self.service.endpoint}/resources/"
+            if not cross_bucket:
+                nexus_path += f"{self.service.organisation}/{self.service.project}"
+            # Try to use the id as it was given
+            if url_base in id_without_query or nexus_path in id:
+                # Use the given id in the request
+                url = id_without_query
+                try:
+                    response = requests.get(
+                        url, params=query_params, headers=self.service.headers
+                    )
+                    response.raise_for_status()
+                except HTTPError as e:
+                    raise RetrievalError(_error_message(e))
+            else:
+                raise RetrievalError(_error_message(er))
+        finally:
             if retrieve_source and not cross_bucket:
 
                 response_metadata = requests.get(
                     url_resource, params=query_params, headers=self.service.headers
                 )
                 response_metadata.raise_for_status()
-            if retrieve_source and cross_bucket:
+            elif retrieve_source and cross_bucket:
```

Comment:
I would add ` and response and "_self" in response.json()` to make response.json() has _self

Reply:
done